### PR TITLE
Fix for issue https://github.com/wayfair-incubator/dagger/issues/41

### PR DESCRIPTION
**Pull request by patkivikram**
## Description

This fixes issue  https://github.com/wayfair-incubator/dagger/issues/41
## Type of Change

- [x] Bug Fix
- [ ] New Feature
- [ ] Breaking Change
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please describe)

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/wayfair-incubator/dagger/blob/main/CONTRIBUTING.md)
- [x] Existing issues have been referenced (where applicable)
- [x] I have verified this change is not present in other open pull requests
- [x] Functionality is documented
- [x] All code style checks pass
- [x] New code contribution is covered by automated tests
- [x] All new and existing tests pass

### CODE REVIEW

File: dagger/tasks/task.py [hunk 1]
```diff
@@ -794,7 +794,7 @@ async def process_event_helper(self, event):  # noqa: C901
         processed_task = False
         if mappings:
             for mapping in mappings:
-                if not mapping or len(mapping) < 2:
+                if not mapping or len(mapping) < 2 or not mapping[1]:
```
[the remark this answers is not in the document]
Comment (patkivikram) [reply dagger/tasks/task.py:797]: its an `or` condition so if length is 1 mapping[1] will not be evaluated

[the remark this answers is not in the document]
Comment (cmwindeknecht) [reply dagger/tasks/task.py:797]: Ah - true, my bad.

**Review by cmwindeknecht — APPROVED** [review]

File: dagger/service/services.py [hunk 2]
```diff
@@ -691,7 +691,7 @@ async def _invoke_store_get_value_for_key_with_timer(self, key: str):
         start_time = self.faust_app.loop.time()
         random: Random = Random()
         random.seed(str(key))
-        partition: int = random.randint(0, Dagger.LOCK_STRIPE_SIZE)  # nosec
+        partition: int = random.randint(0, Dagger.LOCK_STRIPE_SIZE - 1)  # nosec
```
Comment (JeffreyGreenII) [inline dagger/service/services.py:694]: To avoid this nasty `- 1`, you can use `random.randrange(Dagger.LOCK_STRIPE_SIZE)` instead

Comment (patkivikram) [reply dagger/service/services.py:694]: oh yeah! thanks @JeffreyGreenII

**Pull request merged**